modules/rank-eval/src/test/java/org/elasticsearch/index/rankeval/DiscountedCumulativeGainAtTests.java

```diff
@@ -95,12 +97,13 @@ public void testDCGAtSix() throws IOException, InterruptedException, ExecutionEx
     public void testDCGAtSixMissingRatings() throws IOException, InterruptedException, ExecutionException {
         List<RatedDocument> rated = new ArrayList<>();
         int[] relevanceRatings = new int[] { 3, 2, 3};
-        SearchHit[] hits = new InternalSearchHit[6];
+        InternalSearchHit[] hits = new InternalSearchHit[6];
         for (int i = 0; i < 6; i++) {
             if (i < relevanceRatings.length) {
-                rated.add(new RatedDocument(Integer.toString(i), relevanceRatings[i]));
+                rated.add(new RatedDocument(new RatedDocumentKey("index", "type", Integer.toString(i)), relevanceRatings[i]));
             }
             hits[i] = new InternalSearchHit(i, Integer.toString(i), new Text("type"), Collections.emptyMap());
+            hits[i].shard(new SearchShardTarget("testnode", new ShardId("index", "uuid", 0)));
```

Comment:
It adds information to the hit about which index it comes from. Needed since we now check the triplet index/type/doc_id instead of just doc_id